### PR TITLE
library cleanup

### PR DESCRIPTION
**Pull request by bobheadxi**
this PR aims to clean up some issues with this lib:
- it's intended for use as a lib, so I removed `vendor`
- a bunch of fields has been privatized to reduce interface size
- lots of code cleanup
- pubsub stuff removed
- added some misc stuff like test setup, travis, etc.

### CODE REVIEW

File: rtfs.go [hunk 1]
```diff
@@ -13,81 +13,70 @@ import (
...
-func (im *IpfsManager) SetTimeout(time time.Duration) {
-	im.Shell.SetTimeout(time)
+// Close shuts down the manager
+func (im *IpfsManager) Close() error {
+	return im.pubsub.Cancel()
 }
 
-// CreateKeystoreManager is used to create a key store manager for ipfs keys
-// for now it just uses a file system keystore manager
-func (im *IpfsManager) CreateKeystoreManager() error {
```
Comment (bonedaddy) [inline rtfs.go:53]: Looks like this was accidentally removed?

Comment (bonedaddy) [reply rtfs.go:53]: Hmm nevermind it appears that it was an issue with the diff?

**Review by bonedaddy — APPROVED** [review]
Well done :100:

File: rtfs.go (hunk 1 above)
Comment (bobheadxi) [reply rtfs.go:53]: this is intentional - i think its unidiomatic to have a constructor return a "partially instantiated" object, which it does in this case (with an empty keystore manager). It is now provided here:

https://github.com/RTradeLtd/rtfs/blob/cleanup/rtfs.go#L22

This allows you to provide whatever keystore you want, or not provide one at all

Comment (bonedaddy) [reply rtfs.go:53]: Very good change :ok_hand: This will be extremely helpful for the eventual move to a secret management system like vault :fire:

**Pull request merged**